compiler/passes/codegen.cpp

```diff
@@ -200,25 +207,29 @@ static void genGlobalInt32(const char *cname, int value) {
 }
 
 static void
-genClassIDs(Vec<TypeSymbol*> & typeSymbols) {
+genClassIDs(Vec<TypeSymbol*> & typeSymbols, bool isHeader) {
   genComment("Class Type Identification Numbers");
 
   int count=0;
   forv_Vec(TypeSymbol, ts, typeSymbols) {
     if (AggregateType* ct = toAggregateType(ts->type)) {
       if (!isReferenceType(ct) && isClass(ct)) {
-        genGlobalDefClassId(ts->cname, count);
+        genGlobalDefClassId(ts->cname, count, isHeader);
         count++;
       }
     }
   }
 }
 static void
-genFtable(Vec<FnSymbol*> & fSymbols) {
+genFtable(Vec<FnSymbol*> & fSymbols, bool isHeader) {
   GenInfo* info = gGenInfo;
   const char* ftable_name = "chpl_ftable";
   if( info->cfile ) {
     FILE* hdrfile = info->cfile;
+    if(isHeader) {
+      fprintf(hdrfile, "extern chpl_fn_p %s[];\n", ftable_name);
+      return;
+    }
```

Comment:
(I'm not quite sure what making these changes for LLVM would mean at this point, @mppf is the expert on that.  Again, I don't think we should make the LLVM changes right now, but it might be good to capture some brainstorms on that)